torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -0,0 +1,13 @@
+from torch._ops import HigherOrderOperator
+
+
+# Used for wrapping a Triton Kernel
+class TritonKernelWrapperMutation(HigherOrderOperator):
+    def __init__(self):
+        super().__init__("triton_kernel_wrapper_mutation")
+
+    def __call__(self, *args, kernel, grid, **kwargs):
+        kernel[grid](*args, **kwargs)
```

Comment:
Is there a reason why we don't do (self, kernel, grid, args: Tuple, kwargs: Dict) instead? That would be a bit cleaner (what if the triton kernel has a kwarg named grid or kernel?)

Reply:
Because of itertools.partial kernel is passed as kwarg, so it has to come after positional args. I made grid be a positional arg earlier on but @jansel asked me to make it a kwarg as well. So I needed to move them after positional arg list of *args.

It is possible that I misunderstand how positional and kwargs work in python but doing (self, kernel, grid, args: Tuple, kwargs: Dict) gives error saying that kernel and grid are defined twice once as pos and once as kw.

Reply:
Let's say we had a triton kernel that accepted `kernel` argument (unlikely, but just going through the exercise):
```
@triton.jit
def sin_kernel(kernel, out):
    pass
```
and a function like
```
@torch.compile
def f(x):
   out = torch.empty_like(x)
   sin_kernel[grid](kernel=x, out=out)
   return out
```
Will this cause a conflict with the `kernel` argument to the HigherOrderOp? (if not, then I'm satisfied). I expect that the `TritonKernelVariable.call_function` receives kwargs= {"kernel": TensorVariable, "out": TensorVariable}, and then us later invoking triton_kernel_wrapper_mutation with 2 different `kernel`s

Reply:
Also, you can pass a positional arg as kwarg in functools.partials:

<img width="332" alt="image" src="https://github.com/pytorch/pytorch/assets/5652049/262d016c-ec35-4340-9897-02ca65961abe">

Reply:
Yes, this does in fact result in `TypeError: 'Tensor' object is not callable`. Since we are using itertools, we are forced to use kwargs AFAICT. How would we be able to name this something a user cannot write themselves?

Reply:
Oh I see the problem. If we did (kernel, grid, args, kwargs) as the signature, we could pass functools.partial the kernel as a positional arg. Does that work? For example:
```
def f(kernel, grid, *args, **kwargs):
   return (kernel, grid, args, kwargs)

kernel = 1
kf = functools.partial(f, kernel)
print(kf(2, 3, 4, five=5))
# (1, 2, (3, 4), {'five': 5})
```